src/controllers/portfolio/portfolio.ts

```diff
@@ -35,11 +34,28 @@ import EventEmitter from '../eventEmitter/eventEmitter'
 
 // We already know that `results.tokens` and `result.collections` tokens have a balance (this is handled by the portfolio lib).
 // Based on that, we can easily find out which hint tokens also have a balance.
-function getHintsWithBalance(result: PortfolioGetResult): {
+function getHintsWithBalance(
+  result: PortfolioGetResult,
+  keepPinned: boolean,
+  additionalHints: GetOptions['additionalHints'] = []
+): {
   erc20s: Hints['erc20s']
   erc721s: Hints['erc721s']
 } {
-  const erc20s = result.tokens.map((token) => token.address)
+  const erc20s = result.tokens
+    .filter((token) => {
+      return (
+        token.amount > 0n ||
+        additionalHints.includes(token.address) ||
+        // Delete pinned tokens' hints if the user has > 1 non-zero tokens
```

Comment:
I don't think this comment describes correctly what's happening below

Reply:
It does actually. It describes the logic below